Patch by Robert Scheck <robert@fedoraproject.org> for Zarafa >= 7.1.8 which enhances the
current DAgent/Gateway log message by the technical reason why the DAgent/Gateway "socket
error" happened.

--- zcp-7.1.8rc1-43691/spooler/DAgent.cpp		2014-01-21 15:38:53.000000000 +0100
+++ zcp-7.1.8rc1-43691/spooler/DAgent.cpp.strerror	2014-01-30 13:34:56.000000000 +0100
@@ -3184,7 +3184,7 @@
 
 		if (err < 0) {
 			if (errno != EINTR) {
-				g_lpLogger->Log(EC_LOGLEVEL_FATAL, "Socket error!");
+				g_lpLogger->Log(EC_LOGLEVEL_FATAL, "Socket error: %s", strerror(errno));
 				g_bQuit = true;
 				hr = MAPI_E_NETWORK_ERROR;
 			}
--- zcp-7.1.8rc1-43691/gateway/Gateway.cpp		2014-01-21 15:38:53.000000000 +0100
+++ zcp-7.1.8rc1-43691/gateway/Gateway.cpp.strerror	2014-01-30 13:54:47.000000000 +0100
@@ -642,7 +642,7 @@
 		err = select(max(max(ulListenPOP3, ulListenIMAP), max(ulListenPOP3s, ulListenIMAPs)) + 1, &readfds, NULL, NULL, &timeout);
 		if (err < 0) {
 			if (errno != EINTR) {
-				g_lpLogger->Log(EC_LOGLEVEL_FATAL, "Socket error!");
+				g_lpLogger->Log(EC_LOGLEVEL_FATAL, "Socket error: %s", strerror(errno));
 				quit = 1;
 				hr = MAPI_E_NETWORK_ERROR;
 			}
